### PR TITLE
Feat: new custom role resource

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of #498 
In this PR I added the custom_role resource and acceptance tests for the new resource.

### CODE REVIEW

File: env0/resource_custom_role.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					ValidateDiagFunc: NewStringInValidator(allowedCustomRoleTypes),
+				},
+			},
+			"is_default_role": {
+				Type:        schema.TypeBool,
+				Description: "Defaults to 'false'",
+				Optional:    true,
+				Default:     false,
```
Comment (TomerHeber) [inline env0/resource_custom_role.go:67]: For optional boolean it's important to add a "Default".

**Review by GiliFaroEnv0 — COMMENTED** [review]
Looks great!
we double check some stuff on the app end and then I'll approve

**Comment by GiliFaroEnv0** [comment]
[Terraform Provider - Open a GH issue](https://trello.com/c/kubzY8fm/7306-terraform-provider-open-a-gh-issue)

**Review by GiliFaroEnv0 — APPROVED** [review]
FINALLY!!

**Pull request merged**